frontend/src/pages/auth/signup/SignUp.tsx

```diff
@@ -1,12 +1,153 @@
+import "../../../common/styles/button.css";
+import "./styles.css";
+import {useNavigate} from "react-router-dom";
+import {UserCredentials} from "../../../types/userCredentials";
+import {Buttons} from "../../../common/constansts/buttons";
+import {Messages} from "../../../common/constansts/messages";
+import {Roles} from "../../../common/constansts/roles";
+import {UserType} from "../../../types/userType";
 
 const SignUp: React.FC = () => {
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [name, setName] = useState('');
+    const navigate = useNavigate();
+    const [customerChecked, setCustomerChecked] = useState(false);
+    const [merchantChecked, setMerchantChecked] = useState(false);
+    const [agreeChecked, setAgreeChecked] = useState(false);
+
+    const handleCustomerCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setCustomerChecked(event.target.checked);
+        if (event.target.checked) {
+            setMerchantChecked(false);
+        }
+    };
+
+    const handleMerchantCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setMerchantChecked(event.target.checked);
+        if (event.target.checked) {
+            setCustomerChecked(false);
+        }
+    };
+
+    const handleAgreeCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setAgreeChecked(event.target.checked);
+    };
+
+    const handleSignUp = () => {
+        const isSignUpEnabled = email && name && password && (customerChecked || merchantChecked) && agreeChecked;
+        if (isSignUpEnabled) {
+            const userCredentials = createUserCredentials(email, name, password);
+            const userType = getUserType(customerChecked);
+            console.log("userCredentials: " + userCredentials + ", userType: " + userType);
+            navigate(Paths.SignIn);
+        }
+    };
+
+    const createUserCredentials = (name: string, email: string, password: string): UserCredentials => {
+        return {
+            email,
+            name,
+            password
+        }
+    }
+
+    const getUserType = (customerChecked: boolean): UserType => {
+        if (customerChecked) {
+            return UserType.Customer;
+        }
+        return UserType.Merchant;
+    }
+
     return (
-        <p>
-            {Titles.SIGN_UP}
-        </p>
+        <Box className="box">
+            <Box className="box-inner">
+                <Box
+                    component="img"
+                    className="logo"
+                    alt={Titles.APP_NAME}
+                    src={Logo}
+                />
+
+                <TextField
+                    placeholder="Email"
+                    variant="outlined"
+                    value={email}
+                    onChange={(e) => setEmail(e.target.value)}
+                    fullWidth
+                    margin="normal"
+                />
```

Comment:
You are going to use email, password at least 3 times in your app, name 2 times. Let's extract these common components. 

Each component should store it's value state and validation result. Default value should be passed from parent component and be set only once in the beginning. https://react.dev/reference/react/useEffect#examples-dependencies.
I think you can use smth like this 
```typescript
useEffect(() => {
        setEmail(defaultEmail)
    }, []); // empty array means you run this hook only once during first rendering of this component
``` 

Then pass both values to the parent component in your case SignUp using callback during typing. When you receive value (e.g. email or password) and validation result ( e.g. isEmailValid), you update UserCredentials and isValid accordingly in parent component.